service/history/transferQueueActiveProcessor.go

```diff
@@ -649,10 +649,10 @@ func (t *transferQueueActiveProcessorImpl) processSignalExecution(task *persiste
 			},
 			Identity:   common.StringPtr(identityHistoryService),
 			SignalName: common.StringPtr(si.SignalName),
-			Input:      si.Input,
+			Input:      si.Input.Data,
```

Comment:
NOTE: we will pass both Data and Headers for compression. That will be along with step 4 changes(API changes for compression)